src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -110,70 +115,75 @@ export default ajaxCaller(class ShareWorkspaceModal extends Component {
           value: searchValue,
           onChange: v => this.setState({ searchValue: v }),
-          suggestions: _.difference(suggestions, _.map('email', acl)),
+          suggestions: Utils.cond(
+            [searchValueValid && !_.includes(searchValue, aclEmails), () => [searchValue]],
+            [remainingSuggestions.length, () => remainingSuggestions],
+            []
+          ),
           style: { fontSize: 16 }
         })
       ])]),
-      h(FormLabel, ['Role']),
-      div({ style: { display: 'flex', alignItems: 'center', justifyContent: 'space-between' } }, [
-        h(AclInput, {
-          value: { accessLevel, canShare, canCompute },
-          onChange: v => this.setState(v),
-          maxAccessLevel: workspace.accessLevel
-        }),
-        h(ButtonPrimary, {
-          onClick: () => this.addAcl(searchValue),
-          disabled: searchValueInvalid,
-          tooltip: searchValueInvalid && 'Not a valid email address'
-        }, ['Add User'])
-      ]),
-      div({ style: styles.currentCollaboratorsArea }, [
-        div({ style: Style.elements.sectionHeader }, ['Current Collaborators']),
-        ...acl.map(this.renderCollaborator),
+      div({ style: { ...Style.elements.sectionHeader, marginTop: '1rem' } }, ['Current Collaborators']),
+      div({ ref: this.list, style: styles.currentCollaboratorsArea }, [
+        h(Fragment, _.map(this.renderCollaborator, Utils.toIndexPairs(acl))),
         !loaded && centeredSpinner()
       ]),
       updateError && div({ style: { marginTop: '1rem' } }, [
-        div({}, ['An error occurred:']),
+        div(['An error occurred:']),
         updateError
       ]),
       working && spinnerOverlay
     ])
   }
 
-  addAcl(email) {
-    const { acl, accessLevel, canShare, canCompute } = this.state
-    this.setState({ acl: Utils.append({ email, accessLevel, canShare, canCompute, pending: false }, acl), searchValue: '' })
-  }
-
-  renderCollaborator = (aclItem, index) => {
+  renderCollaborator = ([index, aclItem]) => {
     const { email, accessLevel, pending } = aclItem
     const POAccessLevel = 'PROJECT_OWNER'
-    const isPO = accessLevel === POAccessLevel
-    const isMe = email === getUser().email
+    const disabled = accessLevel === POAccessLevel || email === getUser().email
     const { workspace } = this.props
-    const { acl } = this.state
+    const { acl, originalAcl } = this.state
+    const isOld = _.find({ email }, originalAcl)
+    const numAdditions = _.filter(({ email }) => !_.some({ email }, originalAcl), acl).length
 
     return div({
-      style: { display: 'flex', padding: '0.5rem', borderTop: index && `1px solid ${colors.dark(0.4)}` }
+      style: {
+        display: 'flex', alignItems: 'center', borderRadius: 5,
+        padding: '0.5rem 0.75rem', marginBottom: 10,
+        border: `1px solid ${isOld ? colors.dark(0.25) : colors.success(0.5)}`,
+        backgroundColor: isOld ? colors.light(0.2) : colors.success(0.05)
```

Comment:
I'm curious others thoughts but to me this would seem to violate the A11y rule D1: "Is information conveyed by methods other than color alone?" i.e. the information that this user in the list is a newly added one is only identified by this green color (which is not enough of a contrast to count as a lightness difference)

Reply:
Thanks for keeping A11y in mind during the PR! I changed this to a dashed border for the new items. I will run it by Lou to get his thoughts

Reply:
oh that's awesome, really like the look of that thanks!

Reply:
Checked with Lou, he liked the change and updated the mock